pkg/handler/handler_test.go

```diff
@@ -51,8 +51,13 @@ func TestServer(t *testing.T) {
 	})
 	go func() {
 		defer close(serverDone)
-		server.ListenAndServe()
-		cancel()
+		defer cancel()
+
+		if err := server.ListenAndServe(); err != nil {
+			if err != http.ErrServerClosed {
+				t.Errorf("Failed to start test server: %v", err)
```

Comment:
Port 8080 was being used and it was hard to tell that it was happening without these changes